lib/curvature/collector.py

```diff
@@ -204,7 +204,10 @@ def join_ways(self):
             sys.stderr.flush()
 
         for route, route_data in self.routes.iteritems():
-            ways = route_data['ways']
+            # Sort ways by OSM id so that joining always happens in the same order
+            # even if the parser returns them in a different order.
+            ways = sorted(route_data['ways'], key=lambda way: way['id'])
```

Comment:
Would this not break the order of ways? It seems that the constituents would not be connected?

I think we should look at the start and end ways and reverse the ways if the last one has a lower ID